aten/src/ATen/native/quantized/cpu/q_adaavgpool.cpp

```diff
@@ -0,0 +1,225 @@
+          sizeD,
+          isizeH,
+          isizeW,
+          osizeH,
+          osizeW,
+          istrideD,
+          istrideH,
+          istrideW);
+    }
+  });
+}
+
+void adaptive_avg_pool2d_out_template(
+    Tensor& output,
+    Tensor input,
+    std::vector<int64_t> output_shape) {
+  /* sizes */
+  int64_t sizeD = input.size(-3);
+  int64_t isizeH = input.size(-2);
+  int64_t isizeW = input.size(-1);
+  /* strides */
+  int64_t istrideD = input.stride(-3);
+  int64_t istrideH = input.stride(-2);
+  int64_t istrideW = input.stride(-1);
+
+  auto osizeH = output_shape[output_shape.size() - 2];
+  auto osizeW = output_shape[output_shape.size() - 1];
+  int64_t sizeB = output_shape.size() == 3 ? 0 : output_shape[0];
+
+  if (input.dim() == 3 || input.size(0) == 1) {
+    AT_DISPATCH_QINT_TYPES(
+        input.scalar_type(), "quantized_adaptive_avg_pool2d", [&] {
+          auto input_data = input.data_ptr<scalar_t>();
+          auto output_data = output.data_ptr<scalar_t>();
+          adaptive_avg_pool2d_single_out_frame<scalar_t, underlying_t>(
+              input_data,
+              output_data,
+              sizeD,
+              isizeH,
+              isizeW,
+              osizeH,
+              osizeW,
+              istrideD,
+              istrideH,
+              istrideW);
+        });
+  } else {
+    int64_t istrideB = input.stride(-4);
+
+    AT_DISPATCH_QINT_TYPES(
+        input.scalar_type(), "quantized_adaptive_avg_pool2d", [&] {
+          auto input_data = input.data_ptr<scalar_t>();
+          auto output_data = output.data_ptr<scalar_t>();
+          adaptive_avg_pool2d_out_frame<scalar_t, underlying_t>(
+              input_data,
+              output_data,
+              sizeB,
+              sizeD,
+              isizeH,
+              isizeW,
+              osizeH,
+              osizeW,
+              istrideB,
+              istrideD,
+              istrideH,
+              istrideW);
+        });
+  }
+}
+
+std::vector<int64_t> get_output_shape(Tensor input, IntArrayRef output_size) {
+  for (int64_t i = 0; i < input.dim(); i++) {
+    TORCH_CHECK(
+        input.size(i) > 0,
+        "adaptive_avg_pooling2d(): expected input to have non-empty spatial "
+        "dimensions, but input has sizes ",
+        input.sizes(),
+        " with dimension ",
+        i,
+        " being empty");
+  }
+
+  TORCH_CHECK(
+      (input.dim() == 3 || input.dim() == 4),
+      "non-empty 3D or 4D (batch mode) tensor expected for input");
+
+  /* sizes */
+  int64_t sizeD = input.size(-3);
+  const auto osizeH = output_size[0];
+  const auto osizeW = output_size[1];
+
+  /* resize output */
+  std::vector<int64_t> output_shape;
+  int64_t sizeB = 0;
+  if (input.dim() == 3) {
+    output_shape = {sizeD, osizeH, osizeW};
+  } else {
+    sizeB = input.size(-4);
+    output_shape = {sizeB, sizeD, osizeH, osizeW};
+  }
+
+  return output_shape;
+}
+} // namespace
+
+Tensor& quantized_adaptive_avg_pool2d_out(
```

Comment:
Is this used?

Additionally, it doesn't seem to me that this takes into account the stride of the output tensor